example/lib/model/ble_peripheral.dart

```diff
@@ -6,8 +6,23 @@ class BlePeripheral extends Equatable {
   final int rssi;
   final bool isConnected;
 
+  BlePeripheralCategory get category {
+    return KnownBlePeripheralCategory.isSensorTag(name) ? BlePeripheralCategory
+        .sensorTag : BlePeripheralCategory.other;
+  }
+
   BlePeripheral(this.name, this.id, this.rssi, this.isConnected);
 
   @override
   List<Object> get props => [name, id];
+}
+
+enum BlePeripheralCategory { sensorTag, other }
+
+class KnownBlePeripheralCategory {
```

Comment:
This class is not a category, this looks like a utility.

Reply:
Refactored to `BlePeripheralCategoryResolver`.